gtk2_ardour/ardour_ui.cc

```diff
@@ -3575,7 +3599,13 @@ ARDOUR_UI::get_session_parameters (bool quit_on_cancel, bool should_be_new, stri
 			ARDOUR_COMMAND_LINE::session_name = "";
 		}
 	}
+	
+	if (_nsm_switches && ret != 0 ) {
+			/*If there is an error at nsm open or switch, don't loop an error dialog, just quit. */
+			exit (1);
+	}
 
+	_nsm_switches = false;
```

Comment:
Is this the right place to reset `_nsm_switches` ?
There are various ways to load or construct a session and not all are via `get_session_parameters()`. I don't know if that applies to NSM though. But perhaps `ARDOUR_UI::set_session()` is a better place to reset it.

Reply:
Hummm, I don't find where set_session() is called in get_session_parameters().
One thing is sure,  NSM always call get_session_parameters() to open or create a session (see nsm.cc line55). And if ardour isn't launch with NSM, _nsm_switches is never set.

Ardour is set as 'ready' for NSM, once get_session_parameters() is terminate (if there is no errror).

Reply:
set_session() is called whenever the session changes. AFICT it's the only central method that's always called in that case.

Reply:
Ok cool. Just a little question about acronym... What means AFICT ???

Reply:
oops. Sorry, I meant AFAICT -- as far as I can tell.